plugins/techdocs-backend/src/DocsBuilder/builder.ts

```diff
@@ -188,16 +189,9 @@ export class DocsBuilder {
       logStream: this.logStream,
     });
 
-    // Remove Prepared directory since it is no longer needed.
-    // Caveat: Can not remove prepared directory in case of git preparer since the
-    // local git repository is used to get etag on subsequent requests.
-    if (this.preparer instanceof UrlPreparer) {
-      this.logger.debug(
-        `Removing prepared directory ${preparedDir} since the site has been generated`,
-      );
+    if (this.preparer.tidy) {
       try {
-        // Not a blocker hence no need to await this.
-        fs.remove(preparedDir);
+        this.preparer.tidy(preparerResponse);
```

Comment:
The `tidy` code seems to be the same for all the preparer implementations, could you explain why having this logic in the preparers instead of here in the builder?

Reply:
Thats the part I am unsure about. There was a conditional on line 194 "if (this.preparar instanceof UrlPreparer)". What I am doing here is making it possible for a preparer to not tidy, if it chooses. Why it would be neccessary for a preparer to not tidy, I am unsure.

Reply:
Thanks for answering, I asked the team for more context, let's see what they can clarify for us 🙂